expertsystem/data.py

```diff
@@ -122,32 +126,11 @@ class HasQuantumNumbers:  # pylint: disable=too-many-instance-attributes
     g_parity: Optional[Parity] = None
 
 
-@dataclass(frozen=True)
-class HasSpin:
-    """Required to disallow default arguments for `.QuantumState`."""
-
-    spin: Spin
-
-
-@dataclass(frozen=True)
-class HasFloatSpin:
-    """Required to disallow default arguments for `.ParticleQuantumState`."""
-
-    spin: float
-
-
-@dataclass(frozen=True)
-class QuantumState(HasQuantumNumbers, HasSpin):
-    """Contains all quantum numbers unambiguously defining a quantum state."""
-
+class ComplexEnergy:
```

Comment:
I'm still not sure about the naming here. For one, `ComplexEnergy` is not to be used outside this module, but okay, that's Sphinx issues. I think that `HasComplexEnergy` signals its intend better. It cannot be a function that returns a `bool`, because it's PascalCase.

Reply:
Note also that the description of this class is misleading too.

Reply:
I dont understand the naming issue here. Both are PascalCase right. I will try to work with a class decorator here.
```python3
@ComplexEnergy
class ComplexEnergyState():
...
```

Reply:
Oh yes, that would be way cleaner!

Reply:
Btw, not sure what PEP naming conventions are for class decorators, but seeing as `dataclass` is snake_case, I assume it should be `complex_energy`.

Reply:
I couldn't get this to work with a decorator. I guess it should be doable, but is currently beyond my skillset...

Reply:
But overall the idea would be to automatically generate a form of composition for `ComplexEnergy` on the class to decorate.

Reply:
Made an issue for it #157